docs/proposals/04-new-core-gardener-cloud-apis.md

```diff
@@ -0,0 +1,698 @@
+# New `core.gardener.cloud/v1alpha1` APIs required to extract cloud-specific/OS-specific knowledge out of Gardener core
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [`ProviderProfile` resource](#providerprofile-resource)
+    * [`Seed` resource](#seed-resource)
+    * [`SecretBinding` resource](#secretbinding-resource)
+    * [`Quota` resource](#quota-resource)
+    * [`BackupBucket` resource](#backupbucket-resource)
+    * [`BackupEntry` resource](#backupentry-resource)
+    * [`Shoot` resource](#shoot-resource)
+    * [`Plant` resource](#plant-resource)
+
+## Summary
+
+In [GEP-1](https://github.com/gardener/gardener/blob/master/docs/proposals/01-extensibility.md) we have proposed how to (re-)design Gardener to allow providers maintaining their provider-specific knowledge out of the core tree.
+Meanwhile, we have progressed a lot and are about to remove the [`CloudBotanist` interface](https://github.com/gardener/gardener/blob/master/pkg/operation/cloudbotanist/types.go) entirely.
+The only missing aspect that will allow providers to really maintain their code out of the core is to design new APIs.
+
+This proposal describes how the new `Shoot`, `Seed` etc. APIs will be re-designed to cope with the changes made with extensibility.
+We already have the new `core.gardener.cloud/v1alpha1` API group that will be the new default soon.
+
+## Motivation
+
+We want to allow providers to individually maintain their specific knowledge without the necessity to touch the Gardener core code.
+In order to achieve the same, we have to provide proper APIs.
+
+### Goals
+
+* Provide proper APIs to allow providers maintaining their code outside of the core codebase.
+* Do not complicate the APIs for end-users such that they can easily create, delete, and maintain shoot clusters.
+
+### Non-Goals
+
+* Let's try to not split everything up into too many different resources. Instead, let's try to keep all relevant information in the same resources when possible/appropriate.
+
+## Proposal
+
+In GEP-1 we already have proposed a first version for new `CloudProfile` and `Shoot` resources.
+In order to deprecate the existing/old `garden.sapcloud.io/v1beta1` API group (and remove it, eventually) we should move all existing resources to the new `core.gardener.cloud/v1alpha1` API group.
```

Comment:
What about existing fields in the old `v1beta1`? They'll be needed to be present during the migration -
I'll suggest to store them in annotations, so roundtrip conversion works. For example `v1` and `v2` for K8S autoscalling is using this [approach](https://github.com/kubernetes/kubernetes/blob/b81d30a5b29af4c13ce6251b27766fb258178ca9/pkg/apis/autoscaling/v1/conversion.go#L292-L437).

Reply:
The migration is a different story and not covered in this document. We'll figure out what's the best way to achieve backwards compatibility. This GEP only describes how the new APIs shall look like.